hardware_integration/include/isobus/hardware_integration/mcp2515_can_interface.hpp

```diff
@@ -0,0 +1,176 @@
+	/// @param[in] address The address of the register to read
+	/// @param[out] data The data that was read
+	/// @returns If the read was successfull
+	bool read_register(const MCPRegister address, std::uint8_t &data);
+
+	/// @brief read multiple byte register of the mcp2515
+	/// @param[in] address The address of the register to read
+	/// @param[out] data The data that was read
+	/// @param[in] length The length of the data to read
+	/// @returns If the read was successfull
+	bool read_register(const MCPRegister address, std::uint8_t *data, const std::size_t length);
+
+	/// @brief modify a register of the mcp2515
+	/// @param[in] address The address of the register to modify
+	/// @param[in] mask The mask to apply to the register
+	/// @param[in] data The data to write to the register
+	/// @returns If the write was successfull
+	bool modify_register(const MCPRegister address, const std::uint8_t mask, const std::uint8_t data);
+
+	/// @brief write a single byte register of the mcp2515
+	/// @param[in] address The address of the register to write
+	/// @param[in] data The data to write to the register
+	/// @returns If the write was successfull
+	bool write_register(const MCPRegister address, const std::uint8_t data);
+
+	/// @brief write multiple byte register of the mcp2515
+	/// @param[in] address The address of the register to write
+	/// @param[in] data The data to write to the register
+	/// @returns If the write was successfull
+	bool write_register(const MCPRegister address, const std::uint8_t data[], const std::size_t length);
+
+	/// @brief Reset the mcp2515 internally
+	/// @returns If the reset was successfull
+	bool write_reset();
+
+	/// @brief set the mode of the mcp2515
+	/// @param[in] mode The mode to set the mcp2515 to
+	/// @returns If the mode was set successfully
+	bool set_mode(const MCPMode mode);
+
+	/// @brief Read a frame from a buffer on the mcp2515
+	/// @param[in, out] canFrame The frame that was read
+	/// @param[in] ctrl The control register of the buffer to read from
+	/// @param[in] data The data register of the buffer to read from
+	/// @param[in] intf The interrupt flag of the buffer to reset after reading
+	/// @returns If the read was successfull
+	bool read_frame(isobus::HardwareInterfaceCANFrame &canFrame, const MCPRegister ctrl, const MCPRegister data, const std::uint8_t intf);
+
+	/// @brief Write a frame to a buffer on the mcp2515
+	/// @param[in] canFrame The frame to write
+	/// @param[in] ctrl The control register of the buffer to write to
+	/// @param[in] sidh The sidh register of the buffer to write to
+	/// @returns If the write was successfull
+	bool write_frame(const isobus::HardwareInterfaceCANFrame &canFrame, const MCPRegister ctrlRegister, const MCPRegister sidhRegister);
+
+	SPIHardwarePlugin *transactionHandler; ///< The SPI transaction handler
+	const std::uint8_t cfg1, cfg2, cfg3; ///< The configuration values for can and clock speed
```

Comment:
Doxygen complains about this, can you split them up onto separate lines with individual doxygen comments?

Something like:
```
const std::uint8_t cfg1; ///< Comment
const std::uint8_t cfg2; ///< Comment
const std::uint8_t cfg3; ///< Comment
```

Reply:
I'm still seeing it as warnings, did you commit these doxygen changes?
```
mcp2515_can_interface.hpp:173: warning: Member cfg1 (variable) of class MCP2515CANInterface is not documented.
mcp2515_can_interface.hpp:174: warning: Member cfg2 (variable) of class MCP2515CANInterface is not documented.
mcp2515_can_interface.hpp:175: warning: Member cfg3 (variable) of class MCP2515CANInterface is not documented.
```

Reply:
Uhm pretty sure a0166b6f3789b2a9f91e1d34f98abefd86dd6c76 includes this fix. I'll try to run doxygen on my end

Reply:
Oh I see, forgot to put ///> instead of //

Reply:
Should be fixed in https://github.com/ad3154/ISO11783-CAN-Stack/pull/91/commits/6ee216497c6b2752a884cada03438f98ef7a624d